stream.go

```diff
@@ -0,0 +1,185 @@
+// +build !js
+
+package webrtc
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/pion/rtp"
+	"github.com/pion/webrtc/v2/pkg/media"
+)
+
+const (
+	rtpOutboundMTU = 1200
+)
+
+// TrackRTPStream represents a single rtp stream
+type TrackRTPStream struct {
+	mu sync.RWMutex
+
+	// stream id, if rid based it's the rid, it ssrc based it's the ssrc as string
+	id string
+
+	ready bool
+
+	payloadType uint8
+	rid         string
+	ssrc        uint32
+	codec       *RTPCodec
+
+	packetizer rtp.Packetizer
+
+	track *Track
+}
+
+// Ready reports if the stream is ready. If it's a read stream it'll
+// be ready when it's SSRC, RID if rid based, payload are know. If
+// it's a write stream it'll always be ready.
+func (s *TrackRTPStream) Ready() bool {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	return s.ready
+}
+
+// RID gets the RID of the stream. If this isn't a rid based stream
+// it'll return an empty value.
+func (s *TrackRTPStream) RID() string {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	return s.rid
+}
+
+// PayloadType gets the PayloadType of the stream. If this is a rid
+// based stream the payload may be not yet known until a packet is
+// received.
+func (s *TrackRTPStream) PayloadType() uint8 {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	return s.payloadType
+}
+
+// SSRC gets the current SSRC of the stream. If a stream is RID based,
+// this is the currently known ssrc (may be 0 until a packet is
+// received) for the stream rid and may change
```

Comment:
(pedantic) Inconsistent throughout on whether punctuation used at end of doc sentences